### DISCUSSION TITLE
Support for Alfa 802.11ah Halow devices?

### DISCUSSION THREAD

**Discussion by JonahGroendal** [svpcom/wfb-ng#303] [Ideas]
Alfa has an impressive list of 900mhz wifi halow adapters/access points (including a pi hat!) that I think would be amazing hardware targets for this project. 900mhz can penetrate much better than 5.8 or 2.4ghz, and has greater range too.

Of course there would be much quite a bit less bandwidth. Would that be a major issue? Depending on the range, I believe these devices get between 1 and 15Mbps.

What do y'all think, is this feasible? 900mhz digital vtx isn't something I've even seen on commercial devices. Could be a cool niche for this project.

**Comment by svpcom** [comment]
These cards are not 802.11{n,a,ac} and I don't know injection-supported drivers for them

**Reply by JonahGroendal** [reply]
Not sure what I'm looking at here but it seems promising? https://gpo.zugaina.org/net-wireless/nrc7292

Seems to be a way to compile the RTL8812AU/21AU and RTL8814AU driver for the nrc7292 802.11ah chip?

**Comment by themeadery** [comment]
I was looking at this new HaLow hardware from Alfa also and got excited. I love lower frequency options in the RF world. My dreams were suddenly crushed when I was reminded that the channel bandwidths are much lower than 2.4/5.8, but then kept reading their stated data rate specs of 15 Mbps on some to 32.5 Mbps on others. That's enough to send well compressed h265 I think (my h265 IPcam looks good at 8 Mbps h265).

As far as your driver comment I think you might be confused. Or maybe I am confused. I believe these are completely different chipsets and you would no be porting the RTL drivers, you would be modifying whatever new drivers these come with. For example: Newracom™ NRC7292 or Morse Micro MM6108. No way to say if monitor mode or packet injection is capable in hardware or software unless somebody wants to bite the bullet to buy them and start hacking. I already had enough problems testing Intel AX200 and RTL818ftv to be frustrated.

**Reply by ChinaMan28** [reply]
So I have the AsiaRF modules....I have no idea what i'm doing. But i'm willing to send you or someone my set to try and get it to work....

**Reply by themeadery** [reply]
I don't think I'm capable of writing drivers, but thanks for the offer. As my profile says, I have no idea idea what I am doing either, lol. My skills end at writing little Python programs.

**Comment by libc0607** [comment]
I'm trying to let a TXW8301-based 802.11ah module work on OpenIPC. The manufacturer provides Linux drivers and documents (only in Chinese).  See here: https://github.com/libc0607/hgic_ah_fmac  and buy here: https://item.taobao.com/item.htm?id=738230419042  

It seems to support packet injection and monitor mode. The problem is that this driver is FullMAC and does not seem to use the mac80211 framework. After insmod, it looks like a wired eth interface.We need to use some specific APIs (based on procfs) in some tools to control transmission. 

So ... do you have any ideas about adding any abstraction layer to WFB to handle these kinds of modules? Or any better idea?

**Reply by svpcom** [reply]
If you know Chinese and have hardware then you can write libpcap wrapper. Which will incapsulate private api

**Reply by libc0607** [reply]
That's new to me. gonna go through the pcap documents. thanks
